sdk/appcenter-auth/src/main/java/com/microsoft/appcenter/auth/Auth.java

```diff
@@ -461,24 +462,28 @@ private synchronized void initAuthenticationClient(String configurationPayload)
             JSONObject configuration = new JSONObject(configurationPayload);
             String identityScope = configuration.getString(IDENTITY_SCOPE);
             String authorityUrl = null;
+            String type = null;
             JSONArray authorities = configuration.getJSONArray(AUTHORITIES);
             for (int i = 0; i < authorities.length(); i++) {
                 JSONObject authority = authorities.getJSONObject(i);
-                if (authority.optBoolean(AUTHORITY_DEFAULT) && AUTHORITY_TYPE_B2C.equals(authority.getString(AUTHORITY_TYPE))) {
-                    authorityUrl = authority.getString(AUTHORITY_URL);
-                    break;
+                if (authority.optBoolean(AUTHORITY_DEFAULT)) {
+                    if (AUTHORITY_TYPE_B2C.equals(authority.getString(AUTHORITY_TYPE))) {
+                        type = AUTHORITY_TYPE_B2C;
+                        authorityUrl = authority.getString(AUTHORITY_URL);
+                    } else if (AUTHORITY_TYPE_AAD.equals(authority.getString(AUTHORITY_TYPE))) {
+                        type = AUTHORITY_TYPE_AAD;
+                    }
                 }
             }
-            if (authorityUrl != null) {
-
-                /* The remaining validation is done by the library. */
-                mAuthenticationClient = new PublicClientApplication(mContext, getConfigFile());
-                mAuthorityUrl = authorityUrl;
-                mIdentityScope = identityScope;
-                AppCenterLog.info(LOG_TAG, "Auth service configured successfully.");
-            } else {
-                throw new IllegalStateException("Cannot find a b2c authority configured to be the default.");
+            if (type == null) {
+                throw new IllegalStateException("Cannot find a default b2c or aad authority configured to be the default.");
+            } else if (type.equals(AUTHORITY_TYPE_B2C) && authorityUrl == null) {
```

Comment:
<img width="702" alt="Screen Shot 2019-09-09 at 3 10 42 PM" src="https://user-images.githubusercontent.com/623672/64570087-4c095f00-d314-11e9-90e0-49ac9c036277.png">

Android Studio is right, `authority.getString(AUTHORIY_URL)` throws an exception if null.

But in `AuthTest` JSON is mocked. So the setup for `downloadConfigurationWithoutUrl` is wrong.
Accept Android Studio fix and change `downloadConfigurationWithoutUrl` to do

`when(authority.getString("url")).thenThrow(new JSONException("missing"))`

Reply:
agreed. Thank you for explaining offline